app/main/routes.py

```diff
@@ -187,6 +189,32 @@ def record(record_id: uuid.UUID):
     return render_template("record.html", record=file_metadata)
 
 
+@bp.route("/download/<uuid:record_id>")
+@access_token_sign_in_required
+def download_record(record_id: uuid.UUID):
+    file = File.query.get_or_404(record_id)
+
+    validate_body_user_groups_or_404(
+        file.file_consignments.consignment_bodies.Name
+    )
+
+    s3 = boto3.client("s3")
+    bucket = current_app.config["RECORD_BUCKET_NAME"]
+    file_metadata = get_file_metadata(record_id)
+    consignment_reference = file_metadata["consignment"]
+    file_path = file_metadata["file_path"]
+    key = f'{consignment_reference}/{file_path.rstrip("/")}'
+    file_name = file_metadata["file_name"]
+    try:
+        response = Response(
+            s3.get_object(Bucket=bucket, Key=key)["Body"].read(),
+            headers={"Content-Disposition": "attachment;filename=" + file_name},
+        )
+        return response
+    except Exception as e:
```

Comment:
This try except is currently unncessary since we just reraise the error.

We probably want to catch the error and then do abort(404) where `from flask import abort`


Also, can we catch the specific exception that we expect might be raised when making the call to s3 and put the try except around the narrowest bit of code we belive the error to eminate from. e.g.

```
    try:
        s3_object = s3.get_object(Bucket=bucket, Key=key)
    except ClientError, e:
        logger.error(e)
        abort(404)
        

    response = Response(
        s3_object["Body"].read(),
        headers={"Content-Disposition": "attachment;filename=" + file_name},
        )
        return response
```